tests/unittests/distros/test_netconfig.py

```diff
@@ -290,76 +289,92 @@ def __str__(self):
         return self.buffer.getvalue()
 
 
-class TestNetCfgDistroBase(FilesystemMockingTestCase):
-    def setUp(self):
-        super(TestNetCfgDistroBase, self).setUp()
-        self.add_patch("cloudinit.util.system_is_snappy", "m_snappy")
-
-    def _get_distro(self, dname, renderers=None, activators=None):
-        cls = distros.fetch(dname)
-        cfg = copy.deepcopy(settings.CFG_BUILTIN)
-        cfg["system_info"]["distro"] = dname
-        system_info_network_cfg = {}
-        if renderers:
-            system_info_network_cfg["renderers"] = renderers
-        if activators:
-            system_info_network_cfg["activators"] = activators
-        if system_info_network_cfg:
-            cfg["system_info"]["network"] = system_info_network_cfg
-        paths = helpers.Paths({})
-        return cls(dname, cfg.get("system_info"), paths)
-
-    def assertCfgEquals(self, blob1, blob2):
-        b1 = dict(SysConf(blob1.strip().splitlines()))
-        b2 = dict(SysConf(blob2.strip().splitlines()))
-        self.assertEqual(b1, b2)
-        for k, v in b1.items():
-            self.assertIn(k, b2)
-        for k, v in b2.items():
-            self.assertIn(k, b1)
-        for k, v in b1.items():
-            self.assertEqual(v, b2[k])
-
-
-class TestNetCfgDistroFreeBSD(TestNetCfgDistroBase):
-    def setUp(self):
-        super(TestNetCfgDistroFreeBSD, self).setUp()
-        ifs_txt = readResource("netinfo/freebsd-ifconfig-output")
-        with mock.patch(
+def _get_distro(dname, renderers=None, activators=None):
+    cls = distros.fetch(dname)
+    cfg = copy.deepcopy(settings.CFG_BUILTIN)
+    cfg["system_info"]["distro"] = dname
+    system_info_network_cfg = {}
+    if renderers:
+        system_info_network_cfg["renderers"] = renderers
+    if activators:
+        system_info_network_cfg["activators"] = activators
+    if system_info_network_cfg:
+        cfg["system_info"]["network"] = system_info_network_cfg
+    paths = helpers.Paths({})
+    return cls(dname, cfg.get("system_info"), paths)
```

Comment:
A lot of this 'private' function looks like [tests/unittests/distros/__init__.py::_get_distro](https://github.com/canonical/cloud-init/blob/main/tests/unittests/distros/__init__.py#L7). Any chance of reuse there to avoid test helper duplication?

Reply:
I have consolidated both functions, very good suggestion!